### PR TITLE
Zimcheck dashboard

### LINKED ISSUE #78: zimcheck dashboard
All Book entries are created with their respective zimcheck JSON output. Ultimately, we'll use this information in Collections to filter-in/out books.

For the moment, we want to be informed of the status of the ZIM we create. We expect many of them to raise issues but we don't know which ones so it's difficult to act upon.

This ticket thus requires the addition of a new page in the UI with:

- [ ] Condensed stats of zimcheck data: how many books passing, how many don't ; with stats for each kind of errors and maybe the Scrapers that raises them most.

- [ ] a ZIM browser (list with pagination) for the list of all those that don't pass. 

- [x] a zimcheck output visualizer so it's easy an useful instead of the raw JSON output. This Component could be used in Book detail view as well.

### PR DESCRIPTION
**Pull request by fadiga**
Close #78 

Screenshot 2023-01-27 at 12 32 54

### CODE REVIEW

File: frontend/src/views/ZimcheckDashboard.vue [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  created () {
+    const parent = this
+    this.startLoading()
+
+    const url = '/dashboard'
+
+    this.queryAPI('GET', url)
+      .then(function (response) {
+        parent.checkData = response.data.checkData
+        parent.checkTotals = response.data.checkTotals
```
Comment (rgaudin) [inline frontend/src/views/ZimcheckDashboard.vue:188]: Here's a hint on how to get a single, scraper-aware dict from the API as well as having an extendable list of checks

```js
        // list of known check we want to display values for (can be extended)
        const knownChecks = new Map([
          ['checksum', 'error'],
          ['integrity', 'error'],
          ['empty', 'error'],
          ['metadata', 'error'],
          ['favicon', 'error'],
          ['main_page', 'error'],
          ['redundant', 'warning'],
          ['url_internal', 'error'],
          ['url_external', 'error'],
          ['redirect', 'error']
        ])

        parent.checkTotals = new Map()
        knownChecks.forEach((value, key) => parent.checkTotals.set(key, 0))

        parent.scraperData = response.data.checkData
        for (const data of Object.values(scraperData)) {
          for (const [check, value] of Object.entries(data)) {
            parent.checkTotals.set(check, (parent.checkTotals.get(check) || 0) + value)
          }
        }
```

**Review by rgaudin — CHANGES_REQUESTED** [review]
Please check inline comments ; haven't looked at the tests yet

File: backend/src/backend/routes/zimcheck_dashboard.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "redirect",
+    ]
+
+    dict_total_checks = {key: 0 for key in keys}
+    for title in await Title.objects.all():
+        try:
+            last_book = (
+                await Book.objects.filter(title=title)
+                .order_by("-period")
+                .exclude_fields(["title", "counter", "languages", "tags"])
```
[the remark this answers is not in the document]
Comment (fadiga) [reply backend/src/backend/routes/zimcheck_dashboard.py]: These are the fields that can be excluded without causing any issues.

File: backend/tests/test_zimcheck_dashboard.py [hunk 3]
```diff
@@ -1,29 +1,52 @@
...
-from backend.models import Book
+from backend.models import Book, Title
 from backend.routes.zimcheck_dashboard import zim_check_dashboard
 
 
 @pytest.mark.asyncio
-async def test_zim_check_dashboard(client):
-    response = await client.get("/v1/dashboard")
+async def test_zim_check_dashboard(client, book_dict):
+    book_dict["zimcheck"].update(
```
Comment (rgaudin) [inline backend/tests/test_zimcheck_dashboard.py:12]: We want this in a fixture so it's centralized.

File: backend/tests/test_zimcheck_dashboard.py [hunk 4]
```diff
@@ -1,29 +1,52 @@
...
-            "metadata": 0,
-            "favicon": 0,
-            "main_page": 0,
-            "redundant": 0,
-            "url_internal": 0,
-            "url_external": 0,
-            "redirect": 0,
-        },
-    }
+    assert response.json() == {"checkData": {"": {"redundant": 2}}}
```
Comment (rgaudin) [inline backend/tests/test_zimcheck_dashboard.py:49]: Please also test one with a Scraper metadata

**Review by rgaudin — CHANGES_REQUESTED** [review]
Almost ready to merge. Please check the comment regarding the test

**Review by rgaudin — APPROVED** [review]
LGTM

**Pull request merged**